tests/unit/test_agent.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+# pylint: disable=protected-access
+"""Test for agent relations."""
+
+import secrets
+from unittest.mock import MagicMock, PropertyMock
+
+import ops.testing
+import pytest
+from charms.operator_libs_linux.v1 import systemd
+
+import service
+from charm import JenkinsAgentCharm
+
+agent_relation_data = {"url": "http://example.com", "jenkins-agent-0_secret": secrets.token_hex(4)}
+
+
+def test_agent_relation_joined(harness: ops.testing.Harness):
+    """
+    arrange: initialized jenkins-agent charm.
+    act: add relation to the jenkins-k8s charm.
+    assert: The agent set the correct information in the unit's relation databag.
+    """
+    harness.begin()
```

Comment:
Can you double check the formatting? As per our [contributing guide](https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#test-structure), we separate the arrange, act and assert logic by a blank line

Reply:
Thanks, I updated the formatting of the tests